src/extras/module-types.js

```diff
@@ -65,15 +63,46 @@ import { resolveUrl } from '../common.js';
                 setterSources.push('function(m){i[' + key + ']=m}');
               }
             });
-          return new Response(new Blob([
-            'System.register([' + deps.join(',') + '],function(e){var i={};return{setters:[' + setterSources.join(',') +
-            '],execute:function(){return WebAssembly.instantiate(System.wasmModules[' + JSON.stringify(url) +
-            '],i).then(function(m){e(m.exports)})}}})'
-          ], {
+          return new Response(new Blob(['System.register([' + deps.join(',') + '],function(e){var i={};return{setters:[' + setterSources.join(',') + '],execute:function(){return WebAssembly.instantiate(System.wasmModules[' + JSON.stringify(url) + '],i).then(function(m){e(m.exports)})}}})'], {
             type: 'application/javascript'
           }));
         });
-      return res;
-    });
+    }
+  }
+
+  var fetch = systemJSPrototype.fetch;
+  systemJSPrototype.fetch = function (url, options) {
+    return fetch(url, options)
+      .then(function (res) {
+        if (options.passThrough) return res;
+
+        if (!res.ok) return res;
+
+        var assertType = ((options.meta || {}).assert || {}).type;
+        var mime = res.headers.get('content-type');
+        var contentType
+        for (var i = 0; i < contentTypes.length; i++) {
+          var rule = contentTypes[i];
+          if (rule[0].test(mime)) {
+            contentType = rule[1]
+            break
+          }
+        }
+
+        // ensure type match
+        if (assertType && contentType){
+          if(contentType === fallback){
+            contentType = assertType
+          }
+          if (assertType !== contentType){
+            console.warn(process.env.SYSTEM_PRODUCTION ? errMsg('W7') : errMsg('W7', `Module ${url} has an unexpected content type of "${mime}": expected module type ${assertType} got ${contentType}.`));
+            contentType = undefined
+          }
+        }
```

Comment:
This seems fine to me, just watch the spacing of the syntax. Also shorter code is always better for this lib!